components/accept-financial-contributions/FindAHostSearch.tsx

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import { gql, useQuery } from '@apollo/client';
+import { take } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+
+import { API_V2_CONTEXT } from '../../lib/graphql/helpers';
+import { Account, Host } from '../../lib/graphql/types/v2/graphql';
+
+import { Box, Flex } from '../Grid';
+import Loading from '../Loading';
+import MessageBox from '../MessageBox';
+import Pager from '../Pager';
+import { P } from '../Text';
+
+import FeaturedFiscalHostResults from './FeaturedFiscalHostResults';
+import OtherFiscalHostResults from './OtherFiscalHostResults';
+
+function useNonEmptyResultCache<T extends { hosts: { nodes: unknown[] } }>(data: T) {
+  const nonEmptyResult = React.useRef(data);
+  React.useEffect(() => {
+    if (data && data?.hosts?.nodes?.length !== 0) {
+      nonEmptyResult.current = data;
+    }
+  }, [data]);
+
+  return nonEmptyResult.current;
+}
+
+const FindAFiscalHostQuery = gql`
+  query FindAFiscalHostQuery(
+    $tags: [String]
+    $limit: Int
+    $offset: Int
+    $country: [CountryISO]
+    $currency: String
+    $searchTerm: String
+  ) {
+    hosts(
+      tag: $tags
+      limit: $limit
+      offset: $offset
+      tagSearchOperator: OR
+      country: $country
+      currency: $currency
+      searchTerm: $searchTerm
+    ) {
+      totalCount
+      limit
+      offset
+      nodes {
+        id
+        legacyId
+        createdAt
+        settings
+        type
+        name
+        slug
+        description
+        longDescription
+        currency
+        totalHostedCollectives
+        hostFeePercent
+        isTrustedHost
+        location {
+          id
+          country
+        }
+        tags
+      }
+    }
+  }
+`;
+
+const PerPage = 30;
+
+export default function FindAHostSearch(props: {
+  communityTags: string[];
+  selectedCountry: string;
+  selectedCurrency: string;
+  searchTerm: string;
+  collective: Account;
+  onHostApplyClick: (host: Partial<Host>) => void;
+}) {
+  const scrollRef = React.useRef<HTMLDivElement>();
+  const [queryPage, setQueryPage] = React.useState(1);
+
+  React.useEffect(() => {
```

Comment:
Always nice to have a small comment to explain these

```suggestion
  // Return to 1st page when one of the parameters change
  React.useEffect(() => {
```